articles/concepts-grovers.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Theory of Grover's search algorithm
+description: Learn about the theory behind Grover's algorithm.
+author: geduardo
+uid: microsoft.quantum.concepts.grovers
+ms.author: v-edsanc
+ms.date: 12/11/2017
+ms.topic: article
+ms.prod: azure-quantum
+no-loc: ['Q#', '$$v', '$$', "$$", '$', "$", $, $$, '\cdots', 'bmatrix', '\ddots', '\equiv', '\sum', '\begin', '\end', '\sqrt', '\otimes', '{', '}', '\text', '\phi', '\kappa', '\psi', '\alpha', '\beta', '\gamma', '\delta', '\omega', '\bra', '\ket', '\boldone', '\\\\', '\\', '=', '\frac', '\text', '\mapsto', '\dagger', '\to', '\begin{cases}', '\end{cases}', '\operatorname', '\braket', '\id', '\expect', '\defeq', '\variance', '\dd', '&', '\begin{align}', '\end{align}', '\Lambda', '\lambda', '\Omega', '\mathrm', '\left', '\right', '\qquad', '\times', '\big', '\langle', '\rangle', '\bigg', '\Big', '|', '\mathbb', '\vec', '\in', '\texttt', '\ne', '<', '>', '\leq', '\geq', '~~', '~', '\begin{bmatrix}', '\end{bmatrix}', '\_']
+---
+
+# Theory of Grover's search algorithm
+
+In this article you'll find a detailed theoretical explanation of the
+mathematical principles that make Grover's algorithm work.
+
+For a practical implementation of Grover's algorithm to solve mathematical
+problems you can read our [guide to implement Grover's search algorithm](xref:microsoft.quantum.tutorial-qdk.grovers).
+
+## Statement of the problem
+
+Any searching task can be expressed with an abstract function $f(x)$ that accepts search items $x$. If the item $x$ is a solution for the search task, then $f(x)=1$. If the item $x$ isn't a solution, then $f(x)=0$. The search problem consists in finding any item $x_0$ such that $f(x_0)=1$. This is, an item $x_0$ that is a solution to the search problem.
+
+The task that Grover's algorithm aims to solve is, given a classical function $f(x):\\{0,1\\}^n \rightarrow\\{0,1\\}$, find an input $x_0$ for which $f(x_0)=1$.
```

Comment:
```suggestion
The task that Grover's algorithm aims to solve is, for a database of $N$ elements, given a classical function $f(x):\\{0,1, \ldots , N-1 \\} \rightarrow\\{0,1\\}$, find an input $x_0$ for which $f(x_0)=1$.
```

Reply:
I think {0,1}^n is the standard notation for the set of n-tuples comprised of 0s and 1s. And we want to avoid the "database" example since Grover's is not useful for databases.